File: engine/model/src/main/scala/tools/packet/stat/UpdatePetStats.scala
package tools.packet.stat

import client.inventory.MaplePet
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class UpdatePetStats(private var _pets: Array[MaplePet]) extends PacketInput {
  def pets: Array[MaplePet] = _pets

  override def opcode(): SendOpcode = SendOpcode.STAT_CHANGED
}

File: engine/engine-model/src/main/java/tools/packet/stat/UpdatePetStats.java
package tools.packet.stat;

import client.inventory.MaplePet;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record UpdatePetStats(MaplePet[] pets) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.STAT_CHANGED;
   }
}